src/test/commands/slash/slash.ts

```diff
@@ -0,0 +1,46 @@
+import Discord, { CommandInteraction } from "discord.js";
+import { Command, CommandEvent, SlashCommandOption, Sendable } from "../../../command.js";
+import { ButtonSelectOption } from "../../../util/buttonSelects.js";
+
+class Slash extends Command {
+    name = "slash";
+    help = {
+        msg: "This is a help message",
+        usage: "<number>",
+    };
+    slashCommand = true;
+    slashOptions:SlashCommandOption[] = [
+        {
+            name: "user",
+            description: "The user to test",
+            type: "USER",
+        }
+    ]; // this can be left out, default = []
+
+    // Run when the command is called.
+    async run(event: CommandEvent) {
+        const {framework, interaction} = event;
+        if(!event.interaction.isCommand()) return; // required bc typescript funni
```

Comment:
You should type the interaction to be specifically a command interaction so that this isn't required